test/integration/cls.test.js

```diff
@@ -141,6 +141,199 @@ if (current.dialect.supports.transactions) {
       });
     });
 
+    // reason for this test: https://github.com/sequelize/sequelize/issues/12973
+    describe('Model Hook integration', () => {
+      function testHooks({ method, hooks: hookNames, optionPos, execute }) {
+        it(`passes the transaction to hooks {${hookNames.join(',')}} when calling ${method}`, async function () {
+          await this.sequelize.transaction(async transaction => {
+            const hooks = Object.create(null);
+
+            for (const hookName of hookNames) {
+              hooks[hookName] = sinon.spy();
+            }
+
+            for (const [hookName, spy] of Object.entries(hooks)) {
+              this.User[hookName](spy);
+            }
+
+            await Reflect.apply(execute, this, [this.User]);
+
+            const spyMatcher = [];
+            // ignore all arguments until we get to the option bag.
+            for (let i = 0; i < optionPos; i++) {
+              spyMatcher.push(sinon.match.any);
+            }
+
+            // find the transaction in the option bag
+            spyMatcher.push(sinon.match.has('transaction', transaction));
+
+            for (const [hookName, spy] of Object.entries(hooks)) {
+              expect(
+                spy,
+                `hook ${hookName} did not receive the transaction from CLS.`,
+              ).to.have.been.calledWith(...spyMatcher);
+            }
+          });
+        });
+      }
+
+      testHooks({
+        method: 'Model.bulkCreate',
+        hooks: ['beforeBulkCreate', 'beforeCreate', 'afterCreate', 'afterBulkCreate'],
+        optionPos: 1,
+        async execute(User) {
+          await User.bulkCreate([{ name: 'bob' }], { individualHooks: true });
+        },
+      });
+
+      // hooks: ['afterFind'], failing in this but the transaction is coming
+      testHooks({
+        method: 'Model.findAll',
+        hooks: ['beforeFind', 'beforeFindAfterExpandIncludeAll', 'beforeFindAfterOptions', 'afterFind'],
+        optionPos: 0,
+        async execute(User) {
+          await User.findAll();
+        },
+      });
```

Comment:
`afterFind` receives the option bug at `optionPos` 1, simply duplicate this test for `afterFind` and set its optionPos at 1

Reply:
okai